05week/accounts-assignment.js

```diff
@@ -0,0 +1,162 @@
+// =============== Justin Essler ==================== //
+// ============= Week 5.2 Assignment ================ //
+// ============ (instead of checkers) =============== //
+// =========== (on branch: marsMission) ============= //
+
+// at the time of assignment submission the get balance() method is still not working and returns undefined,
+// assignment is already late enough, giving up and will have to come back 
+
+
+// Class Acccount 
+    // constructor 
+    //  - accountNumber: pssed into the constructor
+    //  - ownersName: passed into the constructor
+    //  - list of transaction: starts empty
+    // methods
+    //  - balance(): should loop through all the transactions 
+    //    - and return the current balance
+    //  - addTransaction(payee, amount); 
+    //    - checks to see it the amount is valid 
+    //    - (ie. cannot debit more than the balance)
+
+    console.log("at the top of page");
+
+    class Account {
+        constructor (accountNumber, ownersName, transaction = []) {
+            this.accountNumber = accountNumber;
+            this.ownersName = ownersName;
+            this.transaction = transaction;
+        }
+    
+        balance() {
+            // this method should loop through the list of transactions
+            this.transaction.reduce(function(total, currentValue){
+                return {amount: total.amount + currentValue.amount}
```

Comment:
interesting way to solve the problem.
I would have expected you to map the transacations to just amounts, and then call reduce on the amounts array, but this also works.
Nice job